cnd/src/network/mod.rs

```diff
@@ -128,6 +128,36 @@ impl Swarm {
             local_peer_id,
         })
     }
+
+    /// Takes all the data we know about a 'created' swap.  Executes the
+    /// appropriate communication protocols in order to send data to the peer
+    /// and receive data from the peer (remaining swap parameters) in order to
+    /// finalize this swap.
+    pub fn finalize_created_swap<A, B>(
+        &mut self,
+        swap_id: SwapId,
+        swap_digest: SwapDigest,
+        secret_hash: SecretHash,
+        alpha_identity: A,
+        beta_identity: B,
+    ) -> anyhow::Result<()> {
+        let mut swarm = self.swarm.lock().await;
+        // Assume new NetworkBehaviour is called comit still i.e., the
+        // NetworkBehaviour that implements the new communication protocols.
+
+        // Assumptions:
+        // - This works for Han-Han, Han-HALight, HALight-Han
+
+        // Questions:
+        // - Is this all the data we need for a Han-Herc20 swap?
+        // - How are we going to handle swap protocol pairs, Foo-Bar vs Bar-Foo.
```

Comment:
You will either need to pass enough information into this function so you know this or make more functions per combination.

Reply:
I think this is resolved with the comments in the new version.  Resolving.